posts/migrating-a-large-postgres-db-from-self-managed-ec2-to-rds.md

```diff
@@ -0,0 +1,115 @@
+
+# Pick Your Strategy
+
+Now that we have an in-depth understanding of the size, shape, and mechanisms of our source database, it's time to pick a strategy. Multiple strategies exist and some can even be mixed and matched. Do your research, test your strategy, and if you fail, learn from your mistakes, and don't be afraid to pivot. The goal here is to leverage a strategy that works for your database and also takes into consideration the business requirements. If all of this seems very overwhelming, fear not, AWS MAP partners, like the engineers at Ippon Technologies, are here to guide you or to perform the migration for you!
+
+## Third-Party Tools
+
+There are many third-party tools available to perform database migrations. These tools will utilize some of the strategies described below, but "under the hood". There are several benefits to using third-party tools:
+* Reduced manual effort
+* Improved accuracy
+* Reduced downtime
+* Simplified testing and validation
+* Expert Support
+* Cost-effectiveness
+
+Tools like Amazon's Database Migration Service (DMS) can do a lot of the legwork for you and save valuable time and resources. It's important to acknowledge, however, that some databases are just too large and complex for some of the "one size fits all" solutions that are out there. If you have tried a tool and it has come up short, then you will want to employ a more finely tuned manual strategy.
+
+## Using pg_basebackup
+
+`Pg_basebackup` cannot be used to migrate to AWS’s RDS, but can be used to migrate from one self-managed EC2 instance to a different one. I mention it here simply for thoroughness, as you are likely to come across this method in your research. `pg_basebackup` creates a *physical* backup, at the file system level. This tends to be much faster than the logical backups that I will describe below. The reason this cannot be used to migrate to RDS is that the underlying filesystem is not exposed to the end user when using AWS’s RDS.
+
+It is also worth noting, that on large databases, this method will require a decent length maintenance window, as the database needs to be offline for it to be successful. 
+
+## Using pg_dump and pg_restore
+
+`Pg_dump` and `pg_restore` will create a logical backup of your database. Instead of moving the actual data on the disk, it will use SQL-like statements to completely rebuild the database. There are loads of different options when employing this method. For instance, you can use `pg_dump` to dump your data into .csv files or have the files compressed. Once the data has been dumped onto the hard disk, it can then be loaded into the new database with `pg_restore`.
```

Comment:
```suggestion
`Pg_dump` and `pg_restore` will create a logical backup of your database. Instead of moving the actual data on the disk, it will use SQL-like statements to completely rebuild the database. There are many options when employing this method. For instance, you can use `pg_dump` to dump your data into .csv files or have the files compressed. Once the data has been dumped onto the hard disk, it can then be loaded into the new database with `pg_restore`.
```